addons/purchase/models/purchase.py

```diff
@@ -162,6 +150,13 @@ def create(self, vals):
             vals['name'] = self.env['ir.sequence'].next_by_code('purchase.order') or '/'
         return super(PurchaseOrder, self).create(vals)
 
+    @api.multi
+    def write(self, vals):
+        for order in self:
+            if vals.get('date_planned'):
+                self.mapped('order_line').write({'date_planned': vals.get('date_planned')})
```

Comment:
inside `for loop`, because not use `order `instead of `self`,  or not use `for loop` instead

Reply:
@celm1990 done.